src/ValueParsers/DateFormatParser.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace ValueParsers;
+
+use DataValues\TimeValue;
+
+/**
+ * This parser is in essence the inverse operation of Language::sprintfDate.
+ *
+ * @see Language::sprintfDate
+ *
+ * @since 0.8.1
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class DateFormatParser extends StringValueParser {
+
+	const FORMAT_NAME = 'datetime';
+
+	const OPT_DATE_FORMAT = 'dateFormat';
+	const OPT_DIGIT_TRANSFORM_TABLE = 'digitTransformTable';
+
+	/**
+	 * Must be a two-dimensional array, the first dimension mapping the month's numbers 1 to 12 to
+	 * arrays of month
+	 */
+	const OPT_MONTH_NAMES = 'monthNames';
+
+	public function __construct( ParserOptions $options = null ) {
+		parent::__construct( $options );
+
+		$this->defaultOption( self::OPT_DATE_FORMAT, 'j F Y' );
+		$this->defaultOption( self::OPT_DIGIT_TRANSFORM_TABLE, null );
+		$this->defaultOption( self::OPT_MONTH_NAMES, null );
+	}
+
+	/**
+	 * @see StringValueParser::stringParse
+	 *
+	 * @param string $value
+	 *
+	 * @throws ParseException
+	 * @return TimeValue
+	 */
+	protected function stringParse( $value ) {
+		$format = $this->getDateFormat();
+		$formatLength = strlen( $format );
+		$numberPattern = '[' . $this->getNumberCharacters() . ']';
+		$pattern = '';
+
+		for ( $p = 0; $p < $formatLength; $p++ ) {
+			$code = $format[$p];
+
+			if ( $code === 'x' && $p < $formatLength - 1 ) {
+				$code .= $format[++$p];
+			}
+
+			if ( preg_match( '<^x[ijkmot]$>', $code ) && $p < $formatLength - 1 ) {
```

Comment:
All comments are just notes to myself. I feel that the set of weird characters needs explanation.